### PR TITLE
remove look-behind regexp dependency

### PR DESCRIPTION
**Pull request by tophf**
Fixes https://github.com/openstyles/stylus/issues/1220

Needs a new release for Stylus.

### CODE REVIEW

**Comment by tophf** [comment]
I suggest disabling the failing test which suggests an optimized regexp because it's better to keep the original one as is.

File: lib/parse.js [hunk 1]
```diff
@@ -2,7 +2,10 @@
 
 const {URL} = require('url');
 
-const semverRegex = require('semver-regex');
+/* Inlining semver-regex v2.0.0 here instead of unreliably pinning it in package.json.
+ * This is the last version without look-behind so it can run in older browsers supported by Stylus.
+ * https://github.com/sindresorhus/semver-regex/releases/tag/v2.0.0 */
+const RE_SEMVER = /\bv?(?:0|[1-9]\d*)\.(?:0|[1-9]\d*)\.(?:0|[1-9]\d*)(?:-[\da-z-]+(?:\.[\da-z-]+)*)?(?:\+[\da-z-]+(?:\.[\da-z-]+)*)?\b/ig;
```
Comment (silverwind) [inline lib/parse.js:8]:
```suggestion
const RE_SEMVER = /\bv?(?:0|[1-9]\d*)\.(?:0|[1-9]\d*)\.(?:0|[1-9]\d*)(?:-[\da-z-]+(?:\.[\da-z-]+)*)?(?:\+[\da-z-]+(?:\.[\da-z-]+)*)?\b/ig; // eslint-disable-line
```

Comment (tophf) [reply lib/parse.js]: @silverwind it didn't help, see the details...

**Review by silverwind — APPROVED** [review]

**Comment by silverwind** [comment]
Seems there is a actual test failure.

**Review by silverwind — APPROVED** [review]

**Comment by eight04** [comment]
According to the current spec, any dot-separated digits are allowed. I think we should drop semver and implement our own solution following the spec.

`version-invalid-must-match-full` probably should be removed or be changed to:
```
@version |"0.1.1.1 "
```

**Comment by tophf** [comment]
Someone can do it in another PR...

**Comment by tophf** [comment]
Hmm, I'd say semver mode should be configurable via a parameter because many authors explicitly want to adhere to semver.

**Comment by eight04** [comment]
> Hmm, I'd say semver mode should be configurable via a parameter because many authors explicitly want to adhere to semver.

Do you mean the author of usercss or the developer using this library?

I think this feature should be implemented in downstream software. The goal of this library is to parse usercss metadata. As long as the metadata is valid, it should be successfully parsed.

**Comment by tophf** [comment]
For authors, yes, so they don't need a separate tool. I don't know how this library is used besides Stylus though.

**Comment by decembre** [comment]
When this release should be ready for the update ?

**Comment by eight04** [comment]
I think we should:

1. Drop a patch version that reverts semver-regex to 2.0
2. Drop a minor version that uses [the new validator](https://github.com/openstyles/stylus/blob/9e72784b2a98358db72f7be5d1fe93bb41d54f23/js/meta-parser.js#L8-L13)
3. Modify the release process to remind the dev to run browser test before creating a release.
4. Add github actions that can run browser test automatically. (https://github.com/browser-actions/setup-chrome?)

1)~3) can be done later. I'll open an issue to track 4)

BTW, I've also checked [eslint-plugin-compat](https://github.com/amilajack/eslint-plugin-compat/issues/382) and babel. Neither of them can catch such regexp issues.

**Pull request closed without merging**